server/data_model/src/lib.rs

```diff
@@ -105,23 +105,29 @@ pub struct ImageInformation {
 impl ImageInformation {
     pub fn new(
         image_name: String,
+        image_hash: String,
         tag: String,
         base_image: String,
         run_strs: Vec<String>,
         sdk_version: Option<String>,
     ) -> Self {
-        let mut image_hasher = Sha256::new();
-        image_hasher.update(image_name.clone());
-        image_hasher.update(base_image.clone());
-        image_hasher.update(run_strs.clone().join(""));
-        image_hasher.update(sdk_version.clone().unwrap_or("".to_string())); // Igh.....
+        let mut compat_image_hash: String = "".to_string();
```

Comment:
You need to set this to the incoming image hash I believe.
```suggestion
        let mut compat_image_hash: String = image_hash;
```